lib/client/keystore.go

```diff
@@ -240,6 +271,27 @@ func (fs *FSLocalKeyStore) GetKey(proxyHost string, username string) (*Key, erro
 		return nil, trace.Wrap(err)
 	}
 
+	kubeCerts := make(map[string][]byte)
+	// Optionally, read kube TLS certs.
+	// TODO(awly): unit test this.
+	if clusterName != "" {
```

Comment:
This is introducing a lot of change in the existing client code and adds another task for the function it should not ideally perform. Have you considered adding `GetKubeClusterKey` as an alternative? It is an aggregation of `GetKey` with kube cert lookup:
```go
func GetKubeClusterKey(store *FSLocalKeyStore, proxyHost, username, clusterName string) (*Key, error) {
	key, err := store.GetKey(proxyHost, username)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	dirPath, err := store.dirFor(proxyHost, false)
	if err != nil {
		return nil, trace.Wrap(err)
	}
	kubeCerts, err := ReadKubeCerts(dirPath, clusterName)
	if err != nil {
		store.log.Error(err)
		return nil, trace.Wrap(err)
	}
	key.ClusterName = clusterName
	key.KubeTLSCerts = kubeCerts
	return key, nil
}
```
`ReadKubeCerts` could be tested separately and `GetKubeClusterKey` needn't be tested at all since it solely calls into APIs which are tested.

Reply:
Valid point.
Since we'll be storing database and app certs in a similar way, I changed it to use pluggable options.
WDYT?

Reply:
Well, it is an improvement in that it will isolate the code so it becomes easier to test, yes.
You should consider doing this for other related APIs - e.g. `DeleteKey`.